### DISCUSSION TITLE
xarray.DataArray  Resampling

### DISCUSSION THREAD

**Discussion by epifanio** [pydata/xarray#5079] [Q&A] [answered]
Hi, do you know if `xarray` API for resampling is fully compatible with `pandas` API?

I am trying to resample an `xarray.core.dataarray.DataArray` but I got an error: 

```
index must be monotonic for resampling
```

If I convert the data array to pandas, which returns a `pandas.core.series.Series` I can then resample the same data without errors. 

The code to reproduce the issue it is in the last two cells of the following notebook: 
https://nbviewer.jupyter.org/gist/epifanio/8c028c50789ff5fee590eef3eb78e4ac

Any advice on how to proper use the `xarray.resampling` methods? 
can the 'monotonic issue' caused by the presence of NaN?

**Comment by keewis** [comment, marked as answer]
the issue is that your `time` variable is not sorted. You can fix that using ~`argsort`~ `sortby`:
```python
rawdata = rawdata.sortby("time")
```

I'm not sure why `pandas` works for unsorted indexes. Maybe they automatically sort the indexes before resampling?

**Reply by keewis** [reply]
actually, sorting the data is much easier using `sortby`. I've edited the answer accordingly.
